### DISCUSSION TITLE
Simplifying atom

### DISCUSSION THREAD

**Discussion by larini** [pmndrs/jotai#2673] [Q&A] [answered]
Is it possible to define an atom that executes code when setting its value, without using a second atom ?

    import { atom } from "jotai";

    const sideBarAtom = atom(true);

    const useSidebarAtom = atom(
        (get) => get(sideBarAtom),
        (get, set) => {
        set(sideBarAtom, !get(sideBarAtom));
             //////  alert(get(sideBarAtom));
             ///execute something here  
        },
    );

    export { useSidebarAtom };

**Comment by rothsandro** [comment, marked as answer]
You can just pass the default value as first argument, without a separate atom:

```js
const sidebarAtom = atom(
  true, // default value
  (get, set) => {
    set(sidebarAtom, !get(sidebarAtom));
    // execute something here
});
```
